tests/performance_mocks.py

```diff
@@ -0,0 +1,527 @@
+                                                                14620, 14622, 14624, 14627, 14629, 14631, 14633, 14634,
+                                                                14635, 14637, 14640, 14644, 14645, 14647, 14648, 14652,
+                                                                14654, 14656, 14658, 14659, 14661, 14666, 14668, 14669,
+                                                                14671, 14673, 14677, 14680, 14681, 14682, 14686, 14688,
+                                                                14689, 14693, 14695, 14698, 14700, 14701, 14703, 14709,
+                                                                14711, 14714, 14715, 14716, 14717, 14718, 14719, 14722,
+                                                                14723, 14725, 14727, 14730, 14731, 14732, 14733, 14734,
+                                                                14735, 14736, 14739, 14741, 14743, 14750, 14753, 14754,
+                                                                14755, 14759, 14760, 14761, 14767, 14770, 14771, 14772,
+                                                                14773, 14777, 14781, 14782, 14783, 14788, 14789, 14790,
+                                                                14791, 14792, 14795, 14799, 14800, 14807, 14809, 14810,
+                                                                14813, 14814, 14817, 14819, 14820, 14822, 14823, 14824,
+                                                                14825, 14827, 14828, 14829, 14830, 14832, 14834, 14835,
+                                                                14837, 14838, 14840, 14841, 14843, 14844, 14845, 14848,
+                                                                14849, 14850, 14852, 14853, 14854, 14856, 14857, 14861,
+                                                                14864, 14867, 14868, 14872, 14873, 14874, 14881, 14884,
+                                                                14885, 14886, 14887, 14888, 14889, 14890, 14891, 14898,
+                                                                14899, 14900, 14905, 14906, 14910, 14912, 14915, 14916,
+                                                                14918, 14919, 14921, 14922, 14923, 14924, 14925, 14926,
+                                                                14929, 14931, 14932, 14933, 14934, 14935, 14937, 14939,
+                                                                14941, 14942, 14943, 14947, 14949, 14950, 14958, 14959,
+                                                                14960, 14964, 14967, 14968, 14969, 14970, 14972, 14973,
+                                                                14974, 14975, 14978, 14983, 14984, 14985, 14986, 14989,
+                                                                14991, 14992, 14994, 14997, 14999, 15000, 15002)
+
+
+class NotAPublicKey:
```

Comment:
What's the goal of this mock? and wrt to `serials_which_produce_bytes_which_are_not_viable_as_a_pubkey`.....wh..whh..wwhaatt..is....emmm....waaaat?

Reply:
Hahaha, right.  

So, if you remove this mock, you end up generate some 20k keypairs (validating, signing, delegating, and TLSHosting for each of the 5k Ursulas).  That causes this test to go from taking about 7s total (on my laptop) to about 2min.  So that's obviously unacceptable.

Instead, we generate mock keys in serial, and then, when Alice chooses her 8 Ursulas (out of the 4000 about whom she has learned, out of the 5000 that we have made), she verifies them, and at this stage, we're no longer mocking out, which causes the bytes of the key to be cast back into a real key.

The problem is that, for any mock key string, there are going to be some bytes that aren't properly formatted to be a real key.  So, I have included all of the numbers which ultimately produce bytes which can't be used as a public key.

I know it's awkward, but the only other way that I can think of that we can have a test this fast is to hardcode 20k keypairs into the codebase.